thrift/src/main/java/com/linecorp/armeria/common/thrift/text/StackWalkingCurrentThriftMessageClassFinder.java

```diff
@@ -0,0 +1,102 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =================================================================================================
+package com.linecorp.armeria.common.thrift.text;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class StackWalkingCurrentThriftMessageClassFinder extends AbstractCurrentThriftMessageClassFinder {
+    private final Logger logger =
+            LoggerFactory.getLogger(StackWalkingCurrentThriftMessageClassFinder.class);
+    private final DefaultCurrentThriftMessageClassFinder defaultCurrentThriftClassFinder =
+            new DefaultCurrentThriftMessageClassFinder();
+
+    private final MethodHandle walkMH;
+    private final MethodHandle getInstanceMH;
+    private MethodHandle getClassNameMH;
+
+    StackWalkingCurrentThriftMessageClassFinder() throws Throwable {
+        final MethodHandles.Lookup lookup = MethodHandles.lookup();
+        final MethodType walkMT = MethodType.methodType(Object.class, Function.class);
+        final Class<?> stackWalker = Class.forName("java.lang.StackWalker");
+        walkMH = lookup.findVirtual(stackWalker, "walk", walkMT);
+
+        final MethodType getInstanceMT = MethodType.methodType(stackWalker);
+        getInstanceMH = lookup.findStatic(stackWalker, "getInstance", getInstanceMT);
+
+        final Class<?> stackFrame = Class.forName("java.lang.StackWalker$StackFrame");
+        final MethodType getClassNameMT = MethodType.methodType(String.class);
+        getClassNameMH = lookup.findVirtual(stackFrame, "getClassName", getClassNameMT);
+    }
+
+    @Nullable
+    private final Function<Stream<Object>, Class<?>> walkHandler =  s ->
+            s.map(stackFrame -> {
+                try {
+                    return Class.forName(getClassNameMH.invoke(stackFrame).toString());
+                } catch (Throwable t) {
+                    logger.warn("fail to invoke StackWalker.StackFrame.getDeclaringClass.", t);
+                }
+
+                return null;
+            })
+            .map(AbstractCurrentThriftMessageClassFinder::getMatchedClass)
+            .filter(Objects::nonNull)
+            .findFirst()
+            .orElse(null);
+
+    @Nullable
+    @Override
+    public Class<?> get() {
+        final Class<?> clazz;
+
+        try {
+            final Object sw = getInstanceMH.invoke();
+            clazz = (Class<?>) walkMH.invoke(sw, walkHandler);
+        } catch (Throwable t) {
+            logger.warn("fail to invoke StackWalker.walk. " +
+                        "retry using DefaultCurrentThriftMessageClassFinder", t);
+            return defaultCurrentThriftClassFinder.get();
```

Comment:
I guess we do not need to fallback into the default implementation because it should never fail if this class is run from Java 9+. How about just rethrowing the exception? (wrap with an IllegalStateException if necessary.)